test/contrib/hdf5_test.py

```diff
@@ -0,0 +1,148 @@
+        dfa = target.open()
+        pdt.assert_frame_equal(dfa, self.df)
+
+    def testDateExpression(self):
+        expr = "date >= {:%Y%m%d} & date <= {:%Y%m%d}".format(*(datetime(2013, 1, 1), datetime(2013, 5, 1)))
+        target = Hdf5RowTarget(self.hdf5_file, key="/df", row_expr=expr)
+        dfa = target.open()
+        df = self.df.query(expr)
+        self.assertTrue(target.exists, "Did not evaluate expression correctly:\n{}".format(expr))
+        pdt.assert_frame_equal(dfa, df)
+
+    def testSelectUserSubset(self):
+        expr = "id = [{}]".format(",".join(np.arange(100).astype(str)))
+        target = Hdf5RowTarget(self.hdf5_file, key="/df")
+        dfa = target.open(where=pd.Term(expr))
+        df = self.df[self.df.id.isin(np.arange(100))]
+        self.assertTrue(target.exists, "Did not evaluate expression correctly:\n{}".format(expr))
+        pdt.assert_frame_equal(dfa, df)
+
+    def testWriteMultiIndex(self):
+        target = Hdf5RowTarget(self.hdf5_file_empty, "/df", index_cols=("id", "date"))
+        target.write(self.df)
+        dfa = target.open(where=pd.Term("id > 0 & date >= 20130101"))
+        df = self.df[(self.df.id > 0) & (self.df.date >= "20130101")]
+        pdt.assert_frame_equal(dfa, df)
+
+    def testWriteSingleIndex(self):
+        target = Hdf5RowTarget(self.hdf5_file_empty, "/df", index_cols=("id",))
+        target.write(self.df)
+        dfa = target.open(where=pd.Term("id > 0"))
+        df = self.df[self.df.id > 0]
+        dfa.sort_values("id", inplace=True)
+        df.sort_values("id", inplace=True)
+        # print(set(dfa.id.unique()), set(df.id.unique()) )
+        pdt.assert_frame_equal(dfa, df)
+
+    def testAppend(self):
+        target = Hdf5TableTarget(self.hdf5_file_empty, "/df")
+        target.write(self.df.iloc[:100])
+        target.write(self.df.iloc[100:])
+        dfa = target.open()
+        pdt.assert_frame_equal(dfa, self.df)
+
+    def testChunkRead(self):
+        parts = []
+        target = Hdf5TableTarget(self.hdf5_file, "/df")
+        for df in target.read_chunks(1000):
+            parts.append(df)
+        dfa = pd.concat(parts)
+        pdt.assert_frame_equal(dfa, self.df)
+
+    def testChunkedExprReading(self):
+        parts = []
+        target = Hdf5TableTarget(self.hdf5_file, "/df")
+        for df in target.read_chunks(expr_list=["id = {}".format(i) for i in range(10)]):
+            parts.append(df)
+        dfa = pd.concat(parts)
+        dfa.sort_values(["id", "date"], inplace=True)
+        df = self.df.sort_values(["id", "date"])
+        pdt.assert_frame_equal(dfa, df)
+
+    def testMixedTypeColumn(self):
+        target = Hdf5TableTarget(self.hdf5_file_empty, "/df")
+        self.df.loc[0, "id"] = True
+        self.df.loc[1, "id"] = "somestring"
+        with self.assertRaises(TypeError):
+            target.write(self.df.iloc[:100])
+
+    def tearDown(self):
+        shutil.rmtree(self.dir)
+
+
+if __name__ == "__main__":
```

Comment:
Remove this part please. We run tests with tox+nosetests so this thing is not necessary.